### DISCUSSION TITLE
(ASUS related) wifi options on 192.168.4.1 config page

### DISCUSSION THREAD

**Discussion by magiva** [arendst/Tasmota#16637] [Ideas]
can we have more options on the initial config page (192.168.4.1 wifi config)
such as wifi selection (eg wifi 3 for 11g instead of 11n)
i know i can append /cs? and do it from console but not everyone knows this.
being able to set the static ip and network type is so valuable at that stage.

as asus routers have issues with 11n and esp8266 devices where they sometimes connect but mostly dont, if i dont set wifi 3 for 11g at the begining, my device can connect then not and be stuck being denied by asus router (i have a support ticket with asus also).

seems a simple change with a huge benefit.

**Comment by Jason2866** [comment]
Mhh, imho one of the great benefits of Tasmota is the easy way to get it going without many options to select/choose. Since this problem hits only Asus users, i am not a fan since it litters.
Probably most Asus users will have to do more changes...

**Comment by magiva** [comment]
would still be handy to select the type of wifi that is used, without
defaulting to 11n
to be fair, just a button to get to console would be enough

**Comment by Jason2866** [comment]
@arendst  What do you think about a button to switch to console?

**Reply by arendst** [reply]
Adrian did a rewrite to make initial onboarding as clean as possible. Adding a console button will open up a whole new can of worms letting users perform commands which won't work during onboarding.

Let's conclude to let the Asus users use the "secret" /cs? option to fix their needs.

**Reply by sfromis** [reply]
Instead of opening console, maybe adding that ESP8266 [wifi](https://tasmota.github.io/docs/Commands/#wifi) mode could be helpful when "n" causes failure...

**Reply by magiva** [reply]
> Adrian did a rewrite to make initial onboarding as clean as possible. Adding a console button will open up a whole new can of worms letting users perform commands which won't work during onboarding.
> 
> Let's conclude to let the Asus users use the "secret" /cs? option to fix their needs.

yeah i agree with that

**Comment by Jason2866** [comment]
As fallback mode done by Tasmota itself or presented when connect trys failed?

**Reply by sfromis** [reply]
Having no practical experience with the situation, I have no clear answer, but of course "hidden secrets" may be to much of a hurdle to overcome before even getting started. Simplest might be to just have it on the page, as suggested by the thread starter, but something more automatic sounds nice, if practical.

**Reply by ascillato** [reply]
@magiva 

Tasmota already does that automatically in case you try 2 times to connect. Please, can you try it?

In AP mode the ESP8266 can only performs tests to Wi-Fi networks with 11b and 11g support (hardware limitation of the ESP8266 - ESP32 don't have that limitation). What Tasmota does in ESP8266 is (after you try 2 times the same SSID and PASSWORD) is to save SSID and PASSWORD, turn off the AP mode try to perform a connection using 11b/g/n instead of 11b/g. The wifi command will limit only the maximum mode that the Tasmotized device will offer to the router, meaning that WIFI 3 is Wi-Fi mode 802.11b/g. The default is 4 = Wi-Fi mode 802.11b/g/n and it is up to the router which mode will be used.

- AP Mode can offer to your phone a 11b/g
- AP Mode + STA mode trying to test a network can only use 11b/g
- STA Mode alone (no connection to your phone) can perform connections on 11b/g/n

**Comment by magiva** [comment]
Ive read the documentation on it and ive monitored the connected serial console while it tries and fails over and over again for at least 10 times.

What my experience shows me is that it keeps retrying, im not to know if it changes to a different connection type eg 11b g etc but as it continues to fail i can only assume it stays in 11n.

- i should not also that it this issue seems to be party of a proximity issue, if the esp8266 can see both nodes then it fails, if it see one then it succeeds which is what led this scenario to be eventually successful for it not to be still in the onboarding mode.  This scenario was also seen when devices would join the wifi after swapping my google wifi (the worse) for an asus one

When i then power cycle the number of times to get it back into ap mode, i then go to the console with /cs? and set wifi 3.  It then connects immediately with 11g and continues to connect each time without issue.

As said, i have a ticket with asus to look at this, im not holding my breath on them being that interested but you never know
It would be nice to have an wifi connection type option on the onboarding page but i understand the fears of what more options would create.

So ill continue to use the console to switch it as im currently doing.

**Reply by ascillato** [reply]
So, with your router, while Tasmota is in AP mode which uses 11b/g, it can't connect. You have tried more than once and Tasmota disables the AP and try the connection using STA mode alone in 11b/g/n and can't connect either.
And the workaround you found was, while the ESP8266 is in AP mode using WiFi mode 11b/g, you force Tasmota to use mode 11b/g and then it connects. Seems to me that forcing the mode disables something else in the Arduino Core which makes your router to allow the connection.

So, please correct me. Are your exact steps the following ones?
1-Tasmota in AP mode
2-Go to 192.168.4.1/cs? with your phone
3-type wifi 3
4-go to the main screen
5-Enter the correct SSId and Password.
6-And Tasmota connect at once?

**Comment by magiva** [comment]
my asus uses 11b/g/n but asus has an issue with n, something maybe to do with replicating mac addresses across nodes, maybe wmm, no idea but tasmota doesnt like 11n with asus or asus doesnt li...anyway...
the steps i take.... as previously said, if i go through onboarding as normal (but connected to laptop to monitor console) 
connect to ap mode, configure wifi address, connect..... maybe it connects, maybe it fails, if it connects then onboarding is complete but the connection (11n) is not consistent and following reboots will show it failing failing failing connect failing.  As said, i think this might be if the esp can see more than 1 asus node but im not really sure.  
Which is why i have to power cycle it back to ap mode.
what i do now is.... in onboarding i got to /cs? and set wifi 3, then it doesnt try 11n and connects with 11g perfectly every time.
Tasmota does not, as far as i can see, drop to 11g when 11n fails.  if it did then it would surely connect as it does when wifi 3 is set

**Reply by Jason2866** [reply]
It is a ASUS only issue. I have 6 AP from different vendors, all running in n or ac/n mode with 2.4 and 5G enabled, sending out one (same) SSID. wmm enabled on all AP.  Running OpenWRT on 2 AP. The others are business AP where the orig firmware is running. All AP which support WPA3 have WPA2/3 enabled.
ZERO issues with esp8266 and esp32.

**Comment by magiva** [comment]
ha ha yes i know its an asus only issue.
this thread was to have an option for wifi type selection for onboarding.
it concluded earlier that asus users would use the console to configure

**Comment by Jason2866** [comment]
> Tasmota does not, as far as i can see, drop to 11g when 11n fails. if it did then it would surely connect as it does when wifi 3 is set

This would not help since as you said earlier sometimes it connects with 11n too.

**Comment by magiva** [comment]
as i said.  setting wifi 3 fixes it as it locks it out of 11n.

no further discussion is needed on this particular feature request.
hoping my ticket with asus bears fruit.

**Reply by Jason2866** [reply]
Hope dies last ;-)

**Comment by ascillato** [comment]
What we can do is automate that workaround.

In the onboarding process, after the user set the credentials, Tasmota can try the connection using the AP_STA mode (as it is now) and if it is unsuccessful after let's say 5 attempts, Tasmota can set WiFi mode to 11b/g and try again. So, a change like that won't affect other users. It is just a second layer for testing the connection.

It is quite strange that forcing 11b/g solves the issue for you since in AP_STA mode it is already in that mode. The esp8266 has a hardware limitation that can't support 11b/g/n while in AP_STA mode.

I think it worth to try it. When I have some time I will make that test version and post here for you to test.

**Comment by magiva** [comment]
always happy to help with testing
